sdgym/benchmark.py

```diff
@@ -1115,43 +1163,62 @@ def _add_adjusted_scores(scores, timeout):
     if not uniform_mask.any():
         return _fill_adjusted_scores_with_none(scores)
 
-    uniform_row = scores.loc[uniform_mask].iloc[0]
-    base_fit_time = uniform_row.get('Train_Time')
-    base_sample_time = uniform_row.get('Sample_Time')
-    base_quality_score = uniform_row.get('Quality_Score', None)
-
-    if pd.isna(base_fit_time) or pd.isna(base_sample_time):
-        return _fill_adjusted_scores_with_none(scores)
-
-    fit_times = scores['Train_Time'].fillna(0)
-    sample_times = scores['Sample_Time'].fillna(0)
-    errors = scores.get('error', pd.Series([None] * len(scores)))
-
-    timeout_mask = errors == 'Synthesizer Timeout'
-    other_error_mask = errors.notna() & ~timeout_mask
-    no_error_mask = errors.isna()
-    adjusted_times = np.select(
-        [timeout_mask, other_error_mask, no_error_mask],
-        [
-            base_fit_time + timeout + base_sample_time,
-            base_fit_time + fit_times + sample_times + base_sample_time,
-            base_fit_time + fit_times + sample_times,
-        ],
-        default=np.nan,
-    )
-    scores['Adjusted_Total_Time'] = adjusted_times
-
-    if 'Quality_Score' not in scores.columns:
-        return scores
-
-    if pd.isna(base_quality_score):
-        scores['Adjusted_Quality_Score'] = None
-        return scores
+    scores['Adjusted_Total_Time'] = np.nan
+    if 'Quality_Score' in scores.columns:
+        scores['Adjusted_Quality_Score'] = np.nan
 
-    has_error = errors.notna()
-    scores['Adjusted_Quality_Score'] = np.where(
-        has_error, base_quality_score, scores['Quality_Score']
-    )
+    for dataset in scores['Dataset'].unique():
+        dataset_mask = scores['Dataset'] == dataset
+        uniform_mask = dataset_mask & scores['Synthesizer'].str.contains(
```

Comment:
you do scores['Synthesizer'].str.contains('UniformSynthesizer', na=False) above. Maybe you can store that mask and use it here with the dataset mask and save it as uniform_for_dataset_mask